include/seqan3/alphabet/detail/member_exposure.hpp

```diff
@@ -194,13 +193,55 @@ template <typename alphabet_type>
 //!\cond
     requires requires (alphabet_type alph) { { alph.assign_char(char{0}) } -> alphabet_type &; }
 //!\endcond
-constexpr alphabet_type && assign_char(alphabet_type && alph, underlying_char_t<alphabet_type> const chr) noexcept
+constexpr alphabet_type assign_char(alphabet_type && alph, underlying_char_t<alphabet_type> const chr) noexcept
 {
     static_assert(noexcept(alph.assign_char(chr)),
                   "The assign_char() free function can only forward to .assign_char() member functions that "
                   "are qualified as \"noexcept\".");
     return std::move(alph.assign_char(chr));
```

Comment:
Since we don't require alphabets to not have dynamically allocated memory (they shouldnt though!), this might be a room for optimisation. But plain old `char` cannot be moved anyway.